audio-worklet/src/noisegate-script-processor.js

```diff
@@ -77,20 +79,30 @@ class NoiseGate {
    *                                     input and output buffers
    */
   onaudioprocess_(event) {
+    let inputBuffer = event.inputBuffer;
+    let channel0 = inputBuffer.getChannelData(0);
+
     // Stereo input is downmixed to mono input via averaging.
-    if (event.inputBuffer.numberOfChannels === 2) {
-      for (let i = 0; i < event.inputBuffer.length; i++) {
-        this.channel_[i] = (event.inputBuffer.getChannelData(0)[i] +
-                           event.inputBuffer.getChannelData(1)[i]) / 2;
+    if (inputBuffer.numberOfChannels === 2) {
+      let channel1 = inputBuffer.getChannelData(1);
+      for (let i = 0; i < channel1.length; i++) {
+        this.channel_[i] = (channel0[i] + channel1[i]) / 2;
       }
     } else {
-      this.channel_ = event.inputBuffer.getChannelData(0);
+      this.channel_ = channel0;
     }
 
     let envelope = this.detectLevel_(this.channel_);
 
-    for (let i = 0; i < event.inputBuffer.numberOfChannels; i++) {
-      let input = event.inputBuffer.getChannelData(i);
+    // For sine waves, the envelope eventually reaches an average power of
+    // a^2 / 2. Sine waves are therefore scaled back to the original amplitude,
+    // but other waveforms or constant sources can only be approximated.
+    for (let k = 0; k < this.envelope_.length; k++) {
+      envelope[k] =  this.toDecibel_(Math.sqrt(envelope[k]) * Math.sqrt(2));
+    }
```

Comment:
From a complexity viewpoint, I think it's way cheaper not to convert the envelope to dB here.  In computeGain, convert the threshold to a linear value, adjust for the sqrt(2) and just compare the envelope against the appropriate linear threshold.

But for educational purposes, I guess this is ok if you want to leave it this way.

Actually, I think it makes a lot more sense if computeGain did this itself instead of doing it here.  You should at least move this computation to computeGain if this dB envelope is no longer needed.